File: frontend/src/services/interviewResult.ts
import api from "./api";

export interface CategoryScore {
  category: string;
  average_score: number;
  question_count: number;
}

export interface InterviewResult {
  session_id: number;
  interview_id: number;
  status: string;
  total_questions: number;
  answered_questions: number;
  overall_score?: number | null;
  category_scores: CategoryScore[];
  strengths: string[];
  improvements: string[];
}

export async function getInterviewResult(
  sessionId: number
): Promise<InterviewResult> {
  const response = await api.get(
    `/api/interview-results/${sessionId}`
  );

  return response.data;
}

File: backend/app/schemas/interview_result.py
from typing import Optional

from pydantic import BaseModel


class CategoryScore(BaseModel):
    category: str
    average_score: float
    question_count: int


class InterviewResultResponse(BaseModel):
    session_id: int
    interview_id: int
    status: str
    total_questions: int
    answered_questions: int
    overall_score: Optional[float]

    category_scores: list[CategoryScore]

    strengths: list[str]
    improvements: list[str]